apps/manager_app/auth_card.c

```diff
@@ -0,0 +1,542 @@
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_challenge_query(
+    manager_auth_card_challenge_request_t *challenge_request,
+    manager_auth_card_response_t *resp);
+
+/**
+ * @brief Helper to handle result of auth card.
+ * Handles pairing of card based on result, flow state and card pair requirement
+ *
+ * @param result_query object pointer with result for card auth
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_result_query(
+    manager_auth_card_result_t *result_query);
+
+/**
+ * @brief Helper to parse handle queries for auth card.
+ * Hanlder performs following actions
+ * - Decode host query for manager app
+ * - Verify query is for card auth and type of query
+ * - Call relevant handler for queries
+ *
+ * @param usb_evt object to usb_evt with host query
+ * @param resp object pointer card auth response
+ * @return manager_error_code_t MANAGER_TASK_SUCCESS if successful, else error
+ * code
+ */
+manager_error_code_t handle_auth_card_query(usb_event_t *usb_evt,
+                                            manager_auth_card_response_t *resp);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+auth_card_screen_ctx_t auth_card_ctx = {0};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static manager_error_code_t send_auth_card_response(
+    manager_auth_card_response_t *resp) {
+  if (NULL == resp || 0 == resp->which_response) {
+    return MANAGER_TASK_INVALID_ARGS;
+  }
+
+  // Initialize respose with zero values
+  manager_result_t result = MANAGER_RESULT_INIT_ZERO;
+
+  /* TODO: Update the buffer size when all option files have been updated for
+   * manager app*/
+  uint8_t buffer[MANAGER_AUTH_CARD_RESPONSE_SIZE + 4] = {0};
+  size_t bytes_written = 0;
+
+  // Set auth card valid tags
+  result.which_response = MANAGER_QUERY_AUTH_CARD_TAG;
+
+  // Copy response to pb result struct
+  memcpy(&(result.auth_card), resp, sizeof(manager_auth_card_response_t));
+
+  // Encode result to byte-stream
+  if (false ==
+      encode_manager_result(&result, buffer, sizeof(buffer), &bytes_written)) {
+    // Encoding failed, return false
+    return MANAGER_TASK_ENCODING_FAILED;
+  }
+
+  usb_send_msg(buffer, bytes_written);
+  return MANAGER_TASK_SUCCESS;
+}
+
+static void prepare_card_auth_context(
+    manager_auth_card_initiate_request_t *initiate_request) {
+  ASSERT(NULL != initiate_request);
```

Comment:
I think we can dissolve NULL check in static functions

These checks must be already done by the caller

Reply:
removed